### DISCUSSION TITLE
Перенос записей из VK в OpenVK

### DISCUSSION THREAD

**Discussion by veselcraft** [OpenVK/openvk#1399] [Ideas]
@Zhegal навёл на мысль создать функционал экспорта записей из ВК в ОВК, но с сохранением даты записей. Звучит интересно, почему бы такое не реализовать?

Для этого и особых усилий не нужно. Достаточно движку будет "проанализировать" все записи из паблика, внести какие-то свои изменения и отправлять их в базу данных, с пометкой "было перенесено из ВКонтакте". Вопрос: а как без слишком муторной ебли экспортировать все записи? 

Через экспорт данных ВК? Там информация о пабликах не хранится. Через АПИ? Обязательно нужен для этого ключ. Парсить страницу? Муторно, плюс может капчу потребовать.

В общем, можно поступить так:
1. Сделать импорт записей **по запросу**. Так будет намного легче отсеивать действительно ненужные экспорты, и коммуницировать с API уже человеческим образом. Админам достаточно будет просто запустить какой-либо кроссплатформенный скрипт, тот вставит свой ВКшный токен и высосет посты. Всё красиво оформится в архив, и его будет достаточно экспортировать в ОВК
2. Создать утилиту уже на стороне юзеров: тот точно также будет высасывать посты, всё оформляется в архив, и в ОВК на проверку
3. Полуавтоматический аналог первого пункта: Сделать простую форму для экспорта.

**Comment by celestora** [comment]
@veselcraft я не думаю что такое целесообразно делать в движке, но можно сделать портал где юзер авторизоваться может и перенести своё говно из вк в овк
по желанию данную хуйню можно сделать опенсорце

**Comment by veselcraft** [comment]
> @veselcraft я не думаю что такое целесообразно делать в движке, но можно сделать портал где юзер авторизоваться может и перенести своё говно из вк в овк по желанию данную хуйню можно сделать опенсорце

В двиге - конечно. Можно вообще не лезть в код ОВК (разве что туда пометки добавить о переносе маленькой иконкой), и сделать скрипт для юзера (для экспорта записей в архив), и CLI утилиту, который уже рассасывает архив и экспортирует записи в базу данных и распределяет картинки куда надо
